languages/purescript/exercises/concept/pattern-matching/.docs/introduction.md

```diff
@@ -0,0 +1,44 @@
+An `if/else` expression can be used to conditionally execute logic. PureScript also has another, more powerful way to conditionally execute logic: pattern matching. With pattern matching, a value can be tested against one or more _patterns_. An example of such a pattern is matching a value against a constant (e.g. `1` or `"hello"`).
+
+In Purescript, pattern matching is done through the use of `guards`:
+
+```purescript
+describe number
+  | number == 0 = "Zero"
+  | number == 1 = "One"
+```
+
+**Note:** Notice the lack of an `=` sign after `describe number`.
+
+This may look similar to `switch` statements in other languages.
+
+In some cases, you may want to add an additional condition to a pattern. This is knows as a _guard_ (clause):
+
+```purescript
+describe number
+  | number == 0 = "Zero"
+  | number == 1 = "One"
+  | number == 2 = "Two" -- Extra guard clause
```

Comment:
Am I missing the extra guard clause here? It just looks like an extra condition was added. I suspect this is due to it being a fork of the F# exercise, that first introduced pattern matching against a constant value and then added the `when` guard. But as your example already uses a boolean pattern, they are already guards I think.